flang/docs/FortranStandardsSupport.md

```diff
@@ -0,0 +1,115 @@
+TODOs/Not Yet Implemented messages emitted by the compiler for unimplemented features should be treated as authoritative.
+
+The standards support information is provided as a table with three columns that are self explanatory. The Status column uses
+the letters **P**, **Y**, **N** for the implementation status:
+- **P** : When the implementation is incomplete for a few cases
+- **Y** : When the implementation is complete
+- **N** : When the implementation is absent
+
+Note : No distinction is made between the support in the Parser/Semantics and MLIR or Lowering support.
+
+## Fortran 2023
+See [document](F202X.md) for a brief discussion about the new features in Fortran 2023. The following table summarizes the
+status of all important Fortran 2023 features.
+
+| Feature                                                    | Status | Comments                                                |
+|------------------------------------------------------------|--------|---------------------------------------------------------|
+| Allow longer statement lines and overall statement length  | Y      | |
+| Automatic allocation of lengths of character variables     | N      | |
+| The specifiers typeof and classof                          | N      | |
+| Conditional expressions and arguments                      | N      | |
+| More use of boz constants                                  | P      | All usages other than enum are supported |
+| Intrinsics for extracting tokens from a string             | N      | |
+| Intrinsics for Trig functions that work in degrees         | N      | |
+| Intrinsics for Trig functions that work in half revolutions| N      | |
+| Changes to system_clock                                    | N      | |
+| Changes for conformance with the new IEEE standard         | Y      | |
+| Additional named constants to specify kinds                | Y      | |
+| Extensions for c_f_pointer intrinsic                       | N      | |
+| Procedures for converting between fortran and c strings    | N      | |
+| The at edit descriptor                                     | N      | |
+| Control over leading zeros in output of real values        | N      | |
+| Extensions for Namelist                                    | N      | |
+| Allow an object of a type with a coarray ultimate component to be an array or allocatable | N | |
+| Put with Notify                                            | N      | |
+| Error conditions in collectives                            | N      | |
+| Simple procedures                                          | N      | |
+| Using integer arrays to specify subscripts                 | N      | |
+| Using integer arrays to specify rank and bound of an array | N      | |
+| Using an integer constant to specify rank                  | N      | |
+| Reduction specifier for do concurrent                      | Y      | |
```

Comment:
I think this is still in progress. @ergawy can confirm.  Should the status be `N` or `P`?

Reply:
This is supported for the pure Fortran case, for the `do concurrent` -> OpenMP case, this is not supported yet. So I think @kiranchandramohan is indeed correct here.

Reply:
As @ergawy mentions, this is supported for the sequential Fortran case.
But I understand @rouson 's point that this might cause a user to think that reduction works in a parallel sense. So I am changing it to P (partial).

Reply:
@kiranchandramohan actually I hadn't thought about the difference between serial support and parallel support.  My point was that I didn't even realize that serial support for `reduce` existed yet on the main branch of llvm-project.  I'm building `flang` now and will try it out.  If serial support works, then I think either "Y" or "P" is fine and I leave it up to you to decide.